### PR TITLE
Enable nullability in some ToolStrip members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some ToolStrip members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8570)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 1]
```diff
@@ -1779,7 +1779,7 @@ internal ToolTip ToolTip
                 }
                 else
                 {
-                    toolTip = (ToolTip)Properties.GetObject(ToolStrip.s_propToolTip);
+                    toolTip = (ToolTip)Properties.GetObject(ToolStrip.s_propToolTip)!;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:1782]: Consider using `TryGet` pattern instead of using `Contains`.
https://github.com/dotnet/winforms/blob/766c440e9c209194e930216f7d7811405a44a9a0/src/System.Windows.Forms/src/System/Windows/Forms/PropertyStore.cs#L190. `TryGet` is annotated so that we can get rid of the `!`.

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:1782]: I created https://github.com/dotnet/winforms/pull/8575 to apply this suggestion in more places.

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 2]
```diff
@@ -2217,8 +2217,8 @@ private ToolStripItem GetNextItemHorizontal(ToolStripItem start, bool forward)
                 return null;
             }
 
-            Debug.WriteLineIf(s_selectionDebug.TraceVerbose && (current != -1), "[SelectDBG GetNextToolStripItem] Last selected item was " + ((current != -1) ? DisplayedItems[current].Text : ""));
-            Debug.WriteLineIf(s_selectionDebug.TraceVerbose && (current == -1), "[SelectDBG GetNextToolStripItem] Last selected item was null");
+            Debug.WriteLineIf(s_selectionDebug!.TraceVerbose && (current != -1), "[SelectDBG GetNextToolStripItem] Last selected item was " + ((current != -1) ? DisplayedItems[current].Text : ""));
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:2220]: We are trying to use string interpolation throughout our codebase instead now
```suggestion
            Debug.WriteLineIf(s_selectionDebug!.TraceVerbose && (current != -1), $"[SelectDBG GetNextToolStripItem] Last selected item was {((current != -1) ? DisplayedItems[current].Text : "")}");
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs (hunk 1 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:1782]: @gpetrou  - thank you!

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 3]
```diff
@@ -2208,8 +2208,8 @@ private ToolStripItem GetNextItemHorizontal(ToolStripItem start, bool forward)
                 return null;
             }
 
-            Debug.WriteLineIf(s_selectionDebug.TraceVerbose && (current != -1), "[SelectDBG GetNextToolStripItem] Last selected item was " + ((current != -1) ? DisplayedItems[current].Text : ""));
-            Debug.WriteLineIf(s_selectionDebug.TraceVerbose && (current == -1), "[SelectDBG GetNextToolStripItem] Last selected item was null");
+            Debug.WriteLineIf(s_selectionDebug!.TraceVerbose && (current != -1), $"[SelectDBG GetNextToolStripItem] Last selected item was {((current != -1) ? DisplayedItems[current].Text : string.Empty)}");
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:2211]: I don't think we need to do another check for `current != -1` in the string. We are already checking that in the first param of the `Debug.WriteLineIf`.
```suggestion
            Debug.WriteLineIf(s_selectionDebug!.TraceVerbose && (current != -1), $"[SelectDBG GetNextToolStripItem] Last selected item was {DisplayedItems[current].Text}");
```

**Review by lonitra — COMMENTED** [review]
Small comment on an extra check that we do not need. Otherwise looks good!

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 4]
```diff
@@ -2413,7 +2412,14 @@ internal override Size GetPreferredSizeCore(Size proposedSize)
         internal static Size GetPreferredSizeHorizontal(IArrangedElement container, Size proposedConstraints)
         {
             Size maxSize = Size.Empty;
-            ToolStrip toolStrip = container as ToolStrip;
+            ToolStrip? toolStrip = container as ToolStrip;
+
+            Debug.Assert(toolStrip is not null);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:2417]: why this change? earlier we were throwing null reference and now not. If we do not expect Toolstrip to be null at this place, it would be better to throw it right away.

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 5]
```diff
@@ -2565,7 +2578,7 @@ internal bool GetToolStripState(int flag)
             return (_toolStripState & flag) != 0;
         }
 
-        internal virtual ToolStrip GetToplevelOwnerToolStrip()
+        internal virtual ToolStrip? GetToplevelOwnerToolStrip()
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:2581]: why nullable?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:2581]: We override this in `ToolStripDropDown` and there it is nullable.

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs (hunk 4 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:2417]: I applied a similar change to what we did in https://github.com/dotnet/winforms/pull/8117/files.

**Comment by gpetrou** [comment]
@dreddy-work could we merge this?

**Comment by dreddy-work** [comment]
> @dreddy-work could we merge this?

Not yet. We discussed this. I will be adding details on what we should be doing in the scenarios like this.

**Comment by gpetrou** [comment]
> Not yet. We discussed this. I will be adding details on what we should be doing in the scenarios like this.

@dreddy-work could you please share these details?

**Comment by gpetrou** [comment]
@dreddy-work do you want any more changes to this one?

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 6]
```diff
@@ -2472,7 +2471,7 @@ internal static Size GetPreferredSizeVertical(IArrangedElement container)
         {
             Size maxSize = Size.Empty;
             bool requiresOverflow = false;
-            ToolStrip toolStrip = container as ToolStrip;
+            ToolStrip toolStrip = (container as ToolStrip)!;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:2474]: why not 
```suggestion
            ToolStrip toolStrip = (ToolStrip)container;
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 7]
```diff
@@ -2407,14 +2407,7 @@ internal override Size GetPreferredSizeCore(Size proposedSize)
...
             Size maxSize = Size.Empty;
-            ToolStrip? toolStrip = container as ToolStrip;
-
-            Debug.Assert(toolStrip is not null);
-
-            if (toolStrip is null)
-            {
-                return maxSize;
-            }
+            ToolStrip toolStrip = (container as ToolStrip)!;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:2410]: same here.

**Comment by dreddy-work** [comment]
> @dreddy-work do you want any more changes to this one?

Thanks @gpetrou. Rest look good to me.

**Review by dreddy-work — APPROVED** [review]
